fleetspeak/src/server/https/https_test.go

```diff
@@ -411,3 +440,72 @@ func TestHeaderStreaming(t *testing.T) {
 	bw.Close()
 	resp.Body.Close()
 }
+
+func TestHeaderStreamingChecksum(t *testing.T) {
+	ctx := context.Background()
+
+	clientCertHeader := "ssl-client-cert"
+	clientCertChecksumHeader := "ssl-client-cert-checksum"
+	s, _, addr := makeServer(t, "Streaming", clientCertHeader,
+				 clientCertChecksumHeader, cpb.FrontendMode_HEADER_TLS_CHECKSUM)
+	_, cl, bc, fp := makeClient(t)
+	defer s.Stop()
+
+	br, bw := io.Pipe()
+	go func() {
+		// First exchange - these writes must happen during the http.Client.Do call
+		// below, because the server writes headers at the end of the first message
+		// exchange.
+
+		// Start with the magic number:
+		binary.Write(bw, binary.LittleEndian, magic)
+
+		if _, err := bw.Write(makeWrapped()); err != nil {
+			t.Error(err)
+		}
+	}()
+
+	u := url.URL{Scheme: "https", Host: addr, Path: "/streaming-message"}
+	req, err := http.NewRequest("POST", u.String(), br)
+	req.ContentLength = -1
+	req.Close = true
+	req.Header.Set("Expect", "100-continue")
+
+	cc := url.PathEscape(string(bc))
+	req.Header.Set(clientCertHeader, cc)
+	req.Header.Set(clientCertChecksumHeader, fp)
+	if err != nil {
+		t.Fatal(err)
+	}
+	req = req.WithContext(ctx)
+	resp, err := cl.Do(req)
+	if err != nil {
+		t.Fatalf("Streaming post failed (%v): %v", resp, err)
+	}
+	// Read ContactData for first exchange.
+	body := bufio.NewReader(resp.Body)
+	cd, err := readContact(body)
+	if err != nil {
+		t.Error(err)
```

Comment:
Are we able to proceed if it errors?
In case `readContact` returns `nil` for `cd` when it errors, maybe this should be a `Fatal` instead.
If there are partial results, then keep `Error` (it's best to keep going than stop earlier if we can)

Reply:
Implemented as suggested.